File: app/src/main/java/code/name/monkey/retromusic/adapter/base/MediaEntryViewHolder.java
package code.name.monkey.retromusic.adapter.base;

import android.graphics.Color;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;

import com.google.android.material.card.MaterialCardView;
import com.h6ah4i.android.widget.advrecyclerview.utils.AbstractDraggableSwipeableItemViewHolder;

import code.name.monkey.retromusic.R;

public class MediaEntryViewHolder extends AbstractDraggableSwipeableItemViewHolder
        implements View.OnLongClickListener, View.OnClickListener {

    @Nullable
    public View dragView;

    @Nullable
    public View dummyContainer;

    @Nullable
    public ImageView image;

    @Nullable
    public MaterialCardView imageContainerCard;

    @Nullable
    public FrameLayout imageContainer;

    @Nullable
    public TextView imageText;

    @Nullable
    public MaterialCardView imageTextContainer;

    @Nullable
    public View mask;

    @Nullable
    public AppCompatImageView menu;

    @Nullable
    public View paletteColorContainer;

    @Nullable
    public TextView text;

    @Nullable
    public TextView text2;

    @Nullable
    public TextView time;

    @Nullable
    public TextView title;

    public MediaEntryViewHolder(@NonNull View itemView) {
        super(itemView);
        title = itemView.findViewById(R.id.title);
        text = itemView.findViewById(R.id.text);
        text2 = itemView.findViewById(R.id.text2);

        image = itemView.findViewById(R.id.image);
        time = itemView.findViewById(R.id.time);

        imageText = itemView.findViewById(R.id.imageText);
        imageTextContainer = itemView.findViewById(R.id.imageTextContainer);
        imageContainerCard = itemView.findViewById(R.id.imageContainerCard);
        imageContainer = itemView.findViewById(R.id.imageContainer);

        menu = itemView.findViewById(R.id.menu);
        dragView = itemView.findViewById(R.id.drag_view);
        paletteColorContainer = itemView.findViewById(R.id.paletteColorContainer);
        mask = itemView.findViewById(R.id.mask);
        dummyContainer = itemView.findViewById(R.id.dummy_view);

        if (imageContainerCard != null) {
            imageContainerCard.setCardBackgroundColor(Color.TRANSPARENT);
        }
        itemView.setOnClickListener(this);
        itemView.setOnLongClickListener(this);
    }

    @Nullable
    @Override
    public View getSwipeableContainerView() {
        return null;
    }

    @Override
    public void onClick(View v) {
    }

    @Override
    public boolean onLongClick(View v) {
        return false;
    }

    public void setImageTransitionName(@NonNull String transitionName) {
        itemView.setTransitionName(transitionName);
    /* if (imageContainerCard != null) {
        imageContainerCard.setTransitionName(transitionName);
    }
    if (image != null) {
        image.setTransitionName(transitionName);
    }*/
    }
}

File: app/src/main/java/code/name/monkey/retromusic/adapter/base/MediaEntryViewHolder.kt
package code.name.monkey.retromusic.adapter.base

import android.os.Build
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import code.name.monkey.appthemehelper.ThemeStore
import code.name.monkey.retromusic.R

open class MediaEntryViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener, View.OnLongClickListener {
    override fun onLongClick(v: View?): Boolean {
        return false
    }

    override fun onClick(v: View?) {

    }

    var image: ImageView? = null
    var imageText: TextView? = null
    var title: TextView? = null
    var text: TextView? = null
    var imageContainer: ViewGroup? = null
    var imageContainerCard: CardView? = null
    var menu: View? = null
    var separator: View? = null
    var shortSeparator: View? = null
    var dragView: View? = null
    var paletteColorContainer: View? = null
    var time: TextView? = null
    var recyclerView: RecyclerView? = null
    var playSongs: ImageButton? = null
    var mask: View? = null
    var imageTextContainer: CardView? = null

    init {


        title = view.findViewById(R.id.title)
        text = view.findViewById(R.id.text)

        image = view.findViewById(R.id.image)
        imageContainer = view.findViewById(R.id.image_container)
        imageTextContainer = view.findViewById(R.id.image_text_container)
        imageContainerCard = view.findViewById(R.id.image_container_card)

        imageText = view.findViewById(R.id.image_text)

        menu = view.findViewById(R.id.menu)
        dragView = view.findViewById(R.id.drag_view)

        separator = view.findViewById(R.id.separator)
        shortSeparator = view.findViewById(R.id.short_separator)
        paletteColorContainer = view.findViewById(R.id.palette_color_container)

        time = view.findViewById(R.id.time);
        recyclerView = view.findViewById(R.id.recycler_view)

        mask = view.findViewById(R.id.mask)
        playSongs = view.findViewById(R.id.playSongs)

        view.setOnClickListener(this@MediaEntryViewHolder)
        view.setOnLongClickListener(this@MediaEntryViewHolder)

        imageContainerCard?.setCardBackgroundColor(ThemeStore.primaryColor(itemView.context))

    }

    fun setImageTransitionName(transitionName: String) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && image != null) {
            image!!.transitionName = transitionName
        }
    }
}